app/assets/javascripts/components/course_creator/scoping_methods/petscan_query_builder.jsx

```diff
@@ -0,0 +1,51 @@
+import React from 'react';
+import CategoryAutoCompleteInput from '../../common/ScopingMethods/autocomplete_categories_input';
+import { UPDATE_EXCLUDE_CATEGORIES_PETSCAN, UPDATE_EXCLUDE_TEMPLATES_PETSCAN, UPDATE_INCLUDE_CATEGORIES_PETSCAN, UPDATE_INCLUDE_TEMPLATES_PETSCAN } from '../../../constants/scoping_methods';
+import TemplatesAutoCompleteInput from '../../common/ScopingMethods/autocomplete_templates_input';
+import { useSelector } from 'react-redux';
+
+const PetScanQueryBuilder = () => {
```

Comment:
I testing this with some values I expected to work, but the resulting query has 0 results.

I entered "Wikipedia FemTechNet" a category to include, and "WikiProject Feminism" as a template to include.

I can enter the same values as a Category and a Template separately (not via PetScan) and those categories show a non-zero article count. The Category actually only includes pages in the Wikipedia namespace, while the template is a talk page template. But I would at least have expected the PetScan query to return results based on the template.

I note that the PetScan UI shows 'Intersection' rather than 'Union' for the category part, when I view that query. And I note that only mainspace (the unlabeled one) is selected on the 'Page properties' tab, which is possibly why the template didn't return any results.

I think we need to play around with more combinations of PetScan queries to see where it's likely to go wrong compared with user expectations. Maybe the example I tried could be fixed by include additional options (such as more namespaces)?

Reply:
have pushed a fix for this